pkg/ifaces/registerer.go

```diff
@@ -55,19 +71,54 @@ func (r *Registerer) Subscribe(ctx context.Context) (<-chan Event, error) {
 // IfaceNameForIndex gets the interface name given an index as recorded by the underlying
 // interfaces' informer. It backs up into the net.InterfaceByIndex function if the interface
 // has not been previously registered
-func (r *Registerer) IfaceNameForIndex(idx int) (string, bool) {
+func (r *Registerer) IfaceNameForIndexAndMAC(idx int, mac [6]uint8) (string, bool) {
 	r.m.RLock()
-	name, ok := r.ifaces[idx]
+	macsMap, ok := r.ifaces[idx]
 	r.m.RUnlock()
-	if !ok {
-		iface, err := net.InterfaceByIndex(idx)
-		if err != nil {
-			return "", false
+	if ok {
+		if len(macsMap) == 1 {
+			// No risk of collision, just return entry without checking for MAC
+			for _, name := range macsMap {
+				return name, true
+			}
+		} else if len(macsMap) > 1 {
+			// Several entries, need to disambiguate by MAC
+			name, ok := macsMap[mac]
+			if ok {
+				return name, true
+			}
+			// Not found => before falling back to syscall lookup that is CPU intensive, run this quick ovn optimization:
+			// eth0 & ens5 often collide; MAC starting with 0A:58 should be eth0 and not ens5, but since the MAC captured in flow
+			// doesn't match the actual interface MAC, we'll hardcode that.
+			if mac[0] == 0x0a && mac[1] == 0x58 {
```

Comment:
one last comment in this PR since we don't know how many more exceptions will be needed may I suggest have a slice of those hardcode value so new exceptions will be just adding to teh slice w/o any code changes ?

Reply:
you mean, to have it configurable instead of hardcoded?

Reply:
I made it configurable - so now needs also this operator PR https://github.com/netobserv/network-observability-operator/pull/1565